src/app/modules/convai-checker/perspectiveapi.service.ts

```diff
@@ -130,11 +129,26 @@ export class PerspectiveApiService {
         console.error('No server url specified for a non-direct API call.'
                       + ' Defaulting to current hosted address');
       }
-      const headers = new HttpHeaders();
-      headers.append('Content-Type', 'application/json');
 
-      return this.httpClient.post(
-        serverUrl + '/suggest_score', data, {headers});
+      return this.sendRequest(
+          data, `${serverUrl}/suggest_score`, 'suggestScore');
     }
   }
+
+  private sendRequest<T, R>(data: T, endpoint: string, action: string):
```

Comment:
Add a brief comment here about what the different parameters are (I'm mostly curious about "action", which isn't immediately obvious from the name).

Reply:
Good call. Done.